src/wasm/wasm-validator.cpp

```diff
@@ -237,14 +237,17 @@ void WasmValidator::visitAtomicRMW(AtomicRMW* curr) {
 void WasmValidator::visitAtomicCmpxchg(AtomicCmpxchg* curr) {
   validateMemBytes(curr->bytes, curr->type, curr);
 }
-void WasmValidator::validateMemBytes(uint8_t bytes, WasmType ty, Expression* curr) {
+void WasmValidator::validateMemBytes(uint8_t bytes, WasmType type, Expression* curr) {
+  if (type == unreachable) {
```

Comment:
I think the core issue here is that we use the type when we do `shouldBeEqual(getWasmTypeSize(type), 8U ...`). As a result, when the ptr is unreachable, we don't have anything to compare `bytes` to, so `validateMemBytes` can't do anything.

This is actually an issue with printing such a load as well, we print `(unreachable.load ..)` because we don't know what kind of load it is. Perhaps we should add an extra field to loads, so we keep the type they should have separate from the actual type (which depends on the pointer). That would increase the size of every `load` instruction, though, so I don't know.